lib/streamlit/DeltaGenerator.py

```diff
@@ -226,6 +226,34 @@ class NoValue(object):
     pass
 
 
+class FileUploaderEncodingWarning(StreamlitDeprecationWarning):
+    def __init__(self):
+        msg = self._get_message()
+        config_option = "deprecation.v00_064_showfileUploaderEncoding"
+        super(FileUploaderEncodingWarning, self).__init__(
+            msg=msg, config_option=config_option
+        )
+
+    def _get_message(self):
+        return """
+In the beginning of Q4, we will be deprecating the decoding of uploaded files.
+This means that all files will be returned as binary buffers.
+
+If you are expecting a text buffer in your file processing, we recommend wrapping
+your returned buffer in a `TextIOWrapper`. To learn more about `TextIOWrapper` check
+out the python documentation:
+https://docs.python.org/3/library/io.html#io.TextIOWrapper
+
+```
+import io
+import streamlit as st
+
+uploaded_file = st.file_uploader(...)
+text_io = io.TextIOWrapper(uploaded_file)
```

Comment:
We shouldn't say "Q4", as most people don't know what that means. Also some of the text is overly complex (e.g. "If you are expecting a text buffer in your file processing").

What do you think of simplifying to this:

```
The behavior of `st.file_uploader` will soon change to no longer autodetect the file's encoding. This means that _all files_ will be returned as binary buffers.

This change will go in effect after 2020-MM-DD.

If you require a text buffer, you can future-proof your code right now by wrapping the returned buffer in a [`TextIOWrapper`](https://docs.python.org/3/library/io.html#io.TextIOWrapper), as shown below:

\```
import io

file_buffer = st.file_uploader(...)
text_io = io.TextIOWrapper(file_buffer)
\```
```


(Note: I had to "escape" the triple quotes above to avoid having Markdown parse it in this comment. In the code you should remove the leading slash, of course 😉)

Reply:
Tweaked to October 2020 to provide some flexibility with our release schedule :)

Reply:
Are you saying "after October 2020"? What exactly does that mean though? Isn't that the same as "after 2020-10-31"? 

I prefer the clarity of setting an actual date. Also because the date we choose doesn't mean that _the very next day_ there will be a release that breaks things. So saying 2020-10-31 wouldn't mean "things will break on 2020-11-01", but rather "things _may_ break from 2020-11-01 on".

Reply:
I had set it to "in October 2020" but your point is valid. I'll update and pick a date!